enclave-runtime/Cargo.lock

```diff
@@ -3985,7 +3992,7 @@ name = "sp-externalities"
 version = "0.12.0"
 source = "git+https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.26#e8a7d161f39db70cb27fdad6c6e215cf493ebc3b"
 dependencies = [
- "environmental",
+ "environmental 1.1.3 (registry+https://github.com/rust-lang/crates.io-index)",
```

Comment:
I am having a bit of concerns that still have the original `environmental` macro here, although I'd suppose that we don't use the related code, what do you think?

Reply:
Actually, I have thought about it. I believe it is better to apply a patch here, just to be sure.

Reply:
Problem: `sp-exteranlities` imports the environmental crate as `no_std`. But our special edition crate only allows `sgx` or `std` feature. I shortly tried to make it `no_std` compatible, but didn't succeed yet. Macro magic makes it complicated..

I do agree it's not nice. But I also think if one accesses the `sp-externalities` directly, then something is wrong anyway. So I'm not sure if it's worth investing too much time into that.

Reply:
Yes, I forgot about this potential issue. I think it's fine to leave it as is then.

Reply:
However, patching sp-externalities too would probably remove that dependency?

Reply:
Not possible because of the different naming.

Making the environment crate `no_std` compatible doesn't seem to work. But since runtime-interface is only using externalities in `std` mode, I think it's reasonably safe to go on like this. What do you think?